src/common/ceph_time.h

```diff
@@ -382,10 +382,13 @@ void encode(const std::chrono::time_point<Clock, Duration>& t,
 template<typename Clock, typename Duration>
 void decode(std::chrono::time_point<Clock, Duration>& t,
 	    bufferlist::iterator& p) {
-  uint32_t s, ns;
+  uint32_t s;
+  uint32_t ns;
```

Comment:
@adamemerson i don't quite understand you change. the commit message reads 

> Since we're decoding 32-bit integers, just use uint32_t

we are using `uint32_t`, aren't we?

> and then cast them to what utime_t expects.

and i think
```c++
struct timespec ts = {s, ns};
```
and 
```c++
  struct timespec ts = {
    static_cast<time_t>(s),
    static_cast<long int>(ns)};
```
are equivalent. but the later casts the temp variables explicitly. could you help me understand this commit? thanks.